app/models/group.rb

```diff
@@ -1,3 +1,14 @@
 class Group < ApplicationRecord
   validates :name, presence: true
+  before_create :set_external_id
+
+  def to_param
+    external_id
+  end
+
+private
+
+  def set_external_id
+    self.external_id = SecureRandom.urlsafe_base64(10)
```

Comment:
Whats our best option for generating external ids and do we want pick a standard for other objects which will need it as well, such as Form?

- We want something that's sparse enough that they can't be enumerated by guessing URLs
- They way it's implemented here means we generated a random id, if there is already a Group with that id in the db, a unique error will be thrown when we try and save it but we don't try and solve this. If we choose a random value with enough bytes, we don't have to worry about this happening. It tends to be a higher number than you think due to the [birthday problem](https://en.wikipedia.org/wiki/Birthday_problem)
- This is all internal to the admin, we don't allow clients to generate their own id's at any point so we don't need to overthink coordination
- We don't want URLs to be too long and unwieldy 

`urlsafe_base64` defaults to 16 bytes which is a nice number, from alphabet a-z A-Z 0-9 + / =
[SecureRandom](https://apidock.com/ruby/SecureRandom).[urlsafe_base64](https://apidock.com/ruby/SecureRandom/urlsafe_base64/class) = "b4GOKm4pOYU_-BOXcrUGDg"

Base 58 leaves out like base 64 but leaves out: + / = 0 O I
Defaults to 16 __characters__ long
[SecureRandom](https://apidock.com/rails/SecureRandom).[base58](https://apidock.com/rails/SecureRandom/base58/class) # => "4kUgL2pdQMSCQtjE"

There is a handy calculator here for comparing sizes: https://zelark.github.io/nano-id-cc/

Reply:
This was discussed in tech weekly and with @SamJamCul . We decided that an 8 char long base58 string would be a good place to start and review later on if needed.